pkg/repo/badgerdb/repo.go

```diff
@@ -0,0 +1,65 @@
+package badgerdb
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/badger/v3"
+	"github.com/spf13/viper"
+)
+
+// Repo implements the repo interface.
+type Repo struct {
+	db *badger.DB
+}
+
+type option func(*Repo) error
+
+// WithDefaultDB opens a new Badger database in file system
+// with the path from global config e.g. ENV, flag or config file.
+func WithDefaultDB() option {
+	return func(r *Repo) error {
+		path := viper.GetString("db.badgerdb.path")
+		if path == "" {
+			return fmt.Errorf("WithDefaultDB: viper: empty db.badgerdb.path")
+		}
+
+		var err error
+		r.db, err = newDB(badger.DefaultOptions(path))
+
+		if err != nil {
+			return fmt.Errorf("WithDefaultDB: open badger db: %w", err)
+		}
```

Comment:
keeps err handling closer

```suggestion
		var err error
		
		r.db, err = newDB(badger.DefaultOptions(path))
		if err != nil {
			return fmt.Errorf("WithDefaultDB: open badger db: %w", err)
		}
```